builder/src/main/java/cz/xtf/builder/db/PostgreSQL.java

```diff
@@ -77,14 +211,137 @@ protected String getJDBCConnectionStringPattern() {
 
     @Override
     public Map<String, String> getImageVariables() {
-        Map<String, String> vars = super.getImageVariables();
-        vars.put("POSTGRESQL_MAX_CONNECTIONS", "100");
-        vars.put("POSTGRESQL_SHARED_BUFFERS", "16MB");
-        vars.put("POSTGRESQL_MAX_PREPARED_TRANSACTIONS", "90");
-        // Temporary workaround for https://github.com/sclorg/postgresql-container/issues/297
-        // Increase the "set_passwords.sh" timeout from the default 60s to 300s to give the
-        // PostgreSQL server chance properly to start under high OCP cluster load
-        vars.put("PGCTLTIMEOUT", "300");
+        Map<String, String> vars;
+        if (this.isOfficialImage) {
+            vars = new HashMap<>();
+            vars.put(OFFICIAL_IMAGE_POSTGRESQL_USER_ENV_VAR, getUsername());
+            vars.put(OFFICIAL_IMAGE_POSTGRES_PASSWORD_ENV_VAR, getPassword());
+            vars.put(OFFICIAL_IMAGE_POSTGRESQL_DATABASE_ENV_VAR, getDbName());
+            vars.put("PGDATA", this.pgData);
+        } else {
+            vars = super.getImageVariables();
+            vars.putAll(this.vars);
+        }
         return vars;
     }
+
+    @Override
+    public List<String> getImageArgs() {
+        return args;
+    }
+
+    @Override
+    public String getServiceAccount() {
+        return serviceAccount;
+    }
+
+    public static class PostgreSQLBuilder {
+        private String symbolicName;
+        private String dataDir;
+        private PersistentVolumeClaim pvc;
+        private String username;
+        private String password;
+        private String dbName;
+        private boolean configureEnvironment = true;
+        private boolean withLivenessProbe;
+        private boolean withReadinessProbe;
+        private boolean withStartupProbe;
+        private Map<String, String> vars;
+        private List<String> args;
+        private Supplier<String> deploymentConfigName;
+        private Supplier<String> envVarPrefix;
+        private boolean isOfficialImage = false;
```

Comment:
@mnovak1 so what do you think about keeping support for multiple types of images? In principle are you ok with this?

Reply:
I would be for two different implementations. For compatibility, keep the `PostgreSQL.java` class supporting the RH image and deprecate it. Introduce new classes for different images e.g. `OfficialPostgreSQL` and `RedHatDevServicesPostgreSQL`

Reply:
Thanks Resolving.